charts/studio/values.yaml

```diff
@@ -1231,15 +1231,26 @@ vector-aggregator:
     enabled: true
     storageClass: ""
     accessModes: ["ReadWriteOnce"]
-    size: 64Gi
+    size: 128Gi
 
   # -- Vector command.
   command: [/bin/sh, -c]
 
   # -- Vector arguments.
   args:
     - |-
-      while sleep 60; do find /data/vector/logs -type f -mtime +7 -delete; done &
+      STORAGE=/data/vector
+      MAXIMUM_SIZE_PERCENT=80
+      OLDEST_MODIFICATION_DAYS=5
+
+      while sleep 60; do
+        find "$STORAGE" -type f -mtime +"$OLDEST_MODIFICATION_DAYS" -delete
+        until test $(df "$STORAGE" | awk 'NR==2{print $5}' | tr -d %) -lt "$MAXIMUM_SIZE_PERCENT"; do
+          # HACK: assumes directory structure (single level nesting) and all files prefixed with timestamp
+          rm -f "$(find "$STORAGE" -type f | sort -t/ -k5 | tail -n1)"
+        done
+      done &
+
       exec /usr/local/bin/vector --config-dir /etc/vector/
```

Comment:
same question - what if it exits? 

can we add logs to the whole entry point? also do `set -uex` to see what is happening in case it fails

Reply:
> and Kubernetes will be aware.

we also need to be aware

Reply:
This boils down to a more general problem: we can't observe the observability tool. 😅 

The only way I can think of _guaranteeing_ we notice an outage here, is to monitor the health of Vector from the DataChain `master` pod health check; so if something isn't working (i.e. dead-whatever switch), we get a Slack notification.

Reply:
Monitoring for Vector itself can't be addressed as part of this pull request, though. It has to be addressed as a follow-up.

Reply:
kk, that's fine ... let's do this though before we ship it to customers

also, I would really prefer to be notified if cleanup loop is not working

Reply:
I get your point, but making sure that the cleanup script _works_ in a reliable way isn't neither easy nor elegant.

I would rather try to tackle this from the observability standpoint, and...
1. Monitor the aggregator service to make sure it's running.
2. Monitor storage usage metrics for e.g. less than 2% of free space.